src/others/components/map/Map.tsx

```diff
@@ -49,6 +50,14 @@ export const Map = ({ sourceWithLayer, interactiveLayerIds }: MapProps) => {
           const requestData = features[preferredLayerIndex].properties;
           if (!requestData) return;
           const isRegionPopup = features[preferredLayerIndex].layer.id === 'state-fills';
+          // We don't show region popup when zoomed-in too much.
+          if (isRegionPopup && event.target.getZoom() >= MaxRegionVisibleZoomLevel) {
+            if (popupInfo) {
+              setCursor("auto");
```

Comment:
[nit] Could you just call `handleMouseLeave` here? (Maybe rename it to `closePopover` or something to that effect)

Reply:
done, thanks!